### DISCUSSION TITLE
增加图片角度识别

### DISCUSSION THREAD

**Discussion by tongshiyuan** [RapidAI/TableStructureRec#47] [Ideas]
部分学术文献中的表格是横向展示的，pipeline中或许可以增加一步方向的识别，对横向的表格进行旋转

**Comment by Joker1212** [comment]
这个其实我正在搞一个表格检测的pipeline，因为其实除了角度以外还有真实场景下的透视问题，希望能直接一步到位都解决了，先目标检测再语义分割，最后方向识别

**Reply by tongshiyuan** [reply]
真巧，我最近也在做这一块的处理工作，从PDF中尽可能准确完整的提取表格，已经找到还可以的方法从文件中提取表格为图片，现在正在尝试各种工具和模型将图片信息转化为结构化数据

**Reply by Joker1212** [reply]
emmm,理论上有线表格这个仓库里面的wiredv2准确率非常高，无线表格的话跟百度那个差不多，刚出的paddleX的表格模型应对论文报刊场景准确度还是不错的，后面我们应该也会整合一下弄成onnx

**Reply by Joker1212** [reply]
后续的计划是优化表格分类模型，根据paddleX的模型针对无线表格调优训练，然后整合为识别的pipeline，应该效果比目前还能提升很多

**Reply by tongshiyuan** [reply]
十分期待

**Reply by tongshiyuan** [reply]
我这边已经粗略的实现了角度识别和图像旋转，采用tesseract可以快速识别，并采用cv2旋转

**Reply by Joker1212** [reply]
也欢迎给我们贡献方案哈，如果你愿意的话，目前这个仓库只能解决小角度，边缘清晰的旋转问题

**Reply by tongshiyuan** [reply]
留个邮箱交流一下吧，此外我看到[surya](https://github.com/VikParuchuri/surya/tree/master)这个项目，据称效果很好，不知道能否进行评测集成

**Reply by Joker1212** [reply]
emmm,如果是文档的整体解析方案的话，推荐国内有个pdf-extractor，应该效果更好，但是它就是太复杂了，我们后续也会推出cpu优化的效果很好的文档智能解决方案，邮箱的话，直接用这个吧[xinyijianggo@gmail.com](mailto:xinyijianggo@gmail.com)

**Reply by Joker1212** [reply]
评测的话，后面有时间了搞吧，但是看起来一是它的模型调用要收费，二是对中文的支持好像不够好

**Reply by nissansz** [reply]
> pdf-extractor

这个也是github项目？在哪里看？
